server/chat-commands/moderation.ts

```diff
@@ -621,42 +642,47 @@ exports.commands = {
 
 		let proof = '';
 		let userReason = target;
-		let targetLowercase = target.toLowerCase();
+		const targetLowercase = target.toLowerCase();
 		if (target && (targetLowercase.includes('spoiler:') || targetLowercase.includes('spoilers:'))) {
-			let proofIndex = (targetLowercase.includes('spoilers:') ? targetLowercase.indexOf('spoilers:') : targetLowercase.indexOf('spoiler:'));
-			let bump = (targetLowercase.includes('spoilers:') ? 9 : 8);
+			const proofIndex = (
+				targetLowercase.includes('spoilers:') ? targetLowercase.indexOf('spoilers:') :
+				targetLowercase.indexOf('spoiler:')
+			);
+			const bump = (targetLowercase.includes('spoilers:') ? 9 : 8);
 			proof = `(PROOF: ${target.substr(proofIndex + bump, target.length).trim()}) `;
 			userReason = target.substr(0, proofIndex).trim();
 		}
 
 
 		// Use default time for locks.
-		let duration = week ? Date.now() + 7 * 24 * 60 * 60 * 1000 : (month ? Date.now() + 30 * 24 * 60 * 60 * 1000 : null);
+		const duration = week ? Date.now() + 7 * 24 * 60 * 60 * 1000 : (month ? Date.now() + 30 * 24 * 60 * 60 * 1000 : null);
 		let affected = [];
 
 		if (targetUser) {
-			const ignoreAlts = Punishments.sharedIps.has(targetUser.latestIP);
-			affected = await Punishments.lock(targetUser, duration, targetUser.locked, ignoreAlts, userReason);
+			const ignoreAlts = Punishments.sharedIps.has(targetUser.latestIp);
+			affected = await Punishments.lock(targetUser, duration, targetUser.locked as ID, ignoreAlts, userReason);
```

Comment:
```suggestion
			affected = await Punishments.lock(targetUser, duration, targetUser.locked, ignoreAlts, userReason);
```

It doesn't look like `targetUser.locked` is guaranteed to exist here, so the correct place to fix this is in `Punishments. lock`.

To be clear: The purpose of TypeScript is to find bugs and fix them. Your goal is not to "shut up" the compiler, your goal is to understand what the compiler is trying to tell you, and fix the problem that it's pointing out.